numba/mviewbuf.c

```diff
@@ -1,5 +1,199 @@
+    *ptr = self->wrapped_ptr;
+    return self->wrapped_len;
+}
+
+static Py_ssize_t
+BufferProxyObject_getreadbuf(BufferProxyObject *self, Py_ssize_t index,
+                             const void **ptr) {
+    return BufferProxyObject_getwritebuf(self, index, ptr);
+}
+
+static Py_ssize_t
+BufferProxyObject_getsegcount(BufferProxyObject *self, Py_ssize_t *lenp) {
+    return self->wrapped->ob_type->tp_as_buffer->bf_getsegcount(self->wrapped,
+                                                                lenp);
+}
+
+static Py_ssize_t
+BufferProxyObject_getcharbuf(BufferProxyObject *self, Py_ssize_t index,
+                             const char **ptr) {
+    return BufferProxyObject_getwritebuf(self, index, (const void **) ptr);
+}
+
+static PyBufferProcs BufferProxy_as_buffer = {
+    (readbufferproc)BufferProxyObject_getreadbuf,       /*bf_getreadbuffer*/
+    (writebufferproc)BufferProxyObject_getwritebuf,     /*bf_getwritebuffer*/
+    (segcountproc)BufferProxyObject_getsegcount,        /*bf_getsegcount*/
+    (charbufferproc)BufferProxyObject_getcharbuf,       /*bf_getcharbuffer*/
+    /* new buffer protocol */
+    (getbufferproc)BufferProxyObject_getbuffer,         /*bf_getbuffer*/
+    (releasebufferproc)BufferProxyObject_releasebuffer, /*bf_releasebuffer*/
+};
+#endif
+
+static int
+BufferProxy_init(BufferProxyObject *self, PyObject *args, PyObject *kwds) {
+    if (!PyArg_ParseTuple(args, "O", &(self->wrapped))) {
+        self->wrapped = NULL;
+        return -1;
+    }
+    Py_INCREF(self->wrapped);
+    self->ob_exports = 0;
+#if (PY_MAJOR_VERSION < 3)
+    self->ob_old_exports = 0;
+#endif
+    return 0;
+}
+
+static void
+BufferProxy_dealloc(BufferProxyObject *self) {
+    Py_XDECREF(self->wrapped);
+}
+
+static PyTypeObject BufferProxyType = {
+#if (PY_MAJOR_VERSION < 3)
+    PyObject_HEAD_INIT(NULL)
+    0,                               /*ob_size*/
+#else
+    PyVarObject_HEAD_INIT(NULL, 0)
+#endif
+    "mviewbuf.BufferProxy",          /*tp_name*/
+    sizeof(BufferProxyObject),       /*tp_basicsize*/
+    0,                               /*tp_itemsize*/
+    (destructor)BufferProxy_dealloc, /*tp_dealloc*/
+    0,                               /*tp_print*/
+    0,                               /*tp_getattr*/
+    0,                               /*tp_setattr*/
+    0,                               /*tp_compare*/
+    0,                               /*tp_repr*/
+    0,                               /*tp_as_number*/
+    0,                               /*tp_as_sequence*/
+    0,                               /*tp_as_mapping*/
+    0,                               /*tp_hash */
+    0,                               /*tp_call*/
+    0,                               /*tp_str*/
+    0,                               /*tp_getattro*/
+    0,                               /*tp_setattro*/
+    &BufferProxy_as_buffer,          /*tp_as_buffer*/
+    (Py_TPFLAGS_DEFAULT
+#if PY_MAJOR_VERSION < 3
+     | Py_TPFLAGS_CHECKTYPES
```

Comment:
AFAIK, Py_TPFLAGS_CHECKTYPES is only useful when implementing a numeric type (i.e. the tp_as_number methods).